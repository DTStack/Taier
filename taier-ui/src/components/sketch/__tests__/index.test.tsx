import { act, cleanup, fireEvent, render, renderHook } from '@testing-library/react';
import '@testing-library/jest-dom';
import Sketch, { useSketchRef } from '../';
import React from 'react';

jest.useFakeTimers();

describe('Test Sketch Component', () => {
    beforeEach(() => {
        cleanup();
        document.body.innerHTML = '';
    });

    it('Should match snapshot', async () => {
        await act(async () => {
            render(
                <Sketch
                    header={[
                        'input',
                        'inputWithCondition',
                        'owner',
                        'rangeDate',
                        'datePicker',
                        'select',
                        { name: 'radioGroup' },
                    ]}
                    extra={<button>test</button>}
                    columns={[
                        { dataIndex: 'id', key: 'id' },
                        { dataIndex: 'name', key: 'name' },
                    ]}
                    request={jest.fn().mockResolvedValue({
                        total: 2,
                        data: [
                            {
                                id: 1,
                                name: 'test1',
                            },
                            {
                                id: 2,
                                name: 'test2',
                            },
                        ],
                        polling: false,
                    })}
                />
            );
        });
        expect(document.body).toMatchSnapshot();
    });

    it('Should request again when pagination changed', async () => {
        const fn = jest.fn().mockResolvedValue({
            total: 50,
            data: [
                {
                    id: 1,
                    name: 'test1',
                },
                {
                    id: 2,
                    name: 'test2',
                },
            ],
            polling: false,
        });
        await act(async () => {
            render(
                <Sketch
                    header={['input']}
                    columns={[
                        { dataIndex: 'id', key: 'id' },
                        { dataIndex: 'name', key: 'name' },
                    ]}
                    request={fn}
                />
            );
        });

        await act(async () => {
            fireEvent.click(document.querySelector('.ant-pagination-next')!.firstChild!);
        });

        expect(fn).toBeCalledTimes(2);
    });

    it('Should request again when form field changed', async () => {
        const fn = jest.fn().mockResolvedValue({
            total: 50,
            data: [
                {
                    id: 1,
                    name: 'test1',
                },
                {
                    id: 2,
                    name: 'test2',
                },
            ],
            polling: false,
        });
        const changeFn = jest.fn();
        await act(async () => {
            render(
                <Sketch
                    header={[
                        'input',
                        {
                            name: 'inputWithCondition',
                            props: {
                                formItemProps: {
                                    name: 'jest',
                                },
                            },
                        },
                    ]}
                    columns={[
                        { dataIndex: 'id', key: 'id' },
                        { dataIndex: 'name', key: 'name' },
                    ]}
                    request={fn}
                    onFormFieldChange={changeFn}
                />
            );
        });

        await act(async () => {
            // name field is an insensitive field which won't request again
            fireEvent.change(document.querySelector('input#name')!, { target: { value: '123' } });
        });
        expect(changeFn).toBeCalledTimes(1);
        expect(fn).toBeCalledTimes(1);

        await act(async () => {
            // common field would trigger request
            fireEvent.change(document.querySelector('input#jest')!, { target: { value: '123' } });
        });

        expect(changeFn).toBeCalledTimes(2);
        expect(fn).toBeCalledTimes(2);
        expect(fn.mock.calls[1][0]).toEqual({
            name: '123',
            jest: '123',
            multipleNameSuffix: 'fuzzy',
        });
    });

    it('Should support table with select', async () => {
        const fn = jest.fn().mockResolvedValue({
            total: 50,
            data: [
                {
                    id: 1,
                    name: 'test1',
                },
                {
                    id: 2,
                    name: 'test2',
                },
            ],
            polling: false,
        });
        const selectFn = jest.fn();
        await act(async () => {
            render(
                <Sketch
                    header={['input']}
                    columns={[
                        { dataIndex: 'id', key: 'id' },
                        { dataIndex: 'name', key: 'name' },
                    ]}
                    request={fn}
                    onTableSelect={selectFn}
                />
            );
        });

        act(() => {
            fireEvent.click(document.querySelector('table')!.querySelector('label.ant-checkbox-wrapper')!);
        });

        expect(selectFn).toBeCalled();
    });

    it('Should support expand table', async () => {
        const fn = jest.fn().mockResolvedValue({
            total: 50,
            data: [
                {
                    id: 1,
                    name: 'test1',
                },
                {
                    id: 2,
                    name: 'test2',
                    children: [],
                },
            ],
            polling: false,
        });
        const expandFn = jest.fn().mockResolvedValue([]);
        await act(async () => {
            render(
                <Sketch
                    header={['input']}
                    columns={[
                        { dataIndex: 'id', key: 'id' },
                        { dataIndex: 'name', key: 'name' },
                    ]}
                    request={fn}
                    onExpand={expandFn}
                />
            );
        });

        const expandButtons = document.querySelectorAll(
            'button.ant-table-row-expand-icon.ant-table-row-expand-icon-collapsed'
        );
        // ONLY data with children has expand button
        expect(expandButtons.length).toBe(1);

        await act(async () => {
            fireEvent.click(expandButtons[0]);
        });

        expect(expandFn).toBeCalledWith(true, { id: 2, name: 'test2', children: [] });
    });

    it('Should support polling request', async () => {
        const fn = jest
            .fn()
            .mockResolvedValueOnce({
                total: 50,
                data: [
                    {
                        id: 1,
                        name: 'test1',
                    },
                    {
                        id: 2,
                        name: 'test2',
                        children: [],
                    },
                ],
                polling: true,
            })
            .mockResolvedValue({
                total: 50,
                data: [
                    {
                        id: 1,
                        name: 'test1',
                    },
                    {
                        id: 2,
                        name: 'test2',
                        children: [],
                    },
                ],
                polling: false,
            });

        await act(async () => {
            render(
                <Sketch
                    header={['input']}
                    columns={[
                        { dataIndex: 'id', key: 'id' },
                        { dataIndex: 'name', key: 'name' },
                    ]}
                    request={fn}
                />
            );
        });

        expect(fn).toBeCalledTimes(1);
        expect(fn).toBeCalledWith({ name: undefined }, { current: 1, pageSize: 20 }, {}, undefined);

        await act(async () => {
            jest.runAllTimers();
        });

        expect(fn).toBeCalledTimes(2);
        expect(fn).lastCalledWith({ name: undefined }, { current: 1, pageSize: 20 }, {}, undefined);
    });

    it('Should have ref instance', async () => {
        const fn = jest.fn().mockResolvedValue({
            total: 50,
            data: [
                {
                    id: 1,
                    name: 'test1',
                },
            ],
            polling: false,
        });
        const ref = React.createRef<any>();
        await act(async () => {
            render(
                <Sketch
                    header={['input']}
                    columns={[
                        { dataIndex: 'id', key: 'id' },
                        { dataIndex: 'name', key: 'name' },
                    ]}
                    request={fn}
                    actionRef={ref}
                />
            );
        });

        expect(ref.current.selectedRowKeys).toEqual([]);
        expect(ref.current.selectedRows).toEqual([]);

        act(() => {
            ref.current.setSelectedKeys([1, 2, 3]);
        });
        expect(ref.current.selectedRowKeys).toEqual([1, 2, 3]);

        expect(fn).toBeCalledTimes(1);
        await act(async () => {
            ref.current.submit();
        });
        expect(fn).toBeCalledTimes(2);

        expect(ref.current.form.getFieldsValue).toBeInstanceOf(Function);

        expect(ref.current.getTableData()).toEqual([
            {
                id: 1,
                name: 'test1',
            },
        ]);
    });

    it('Should get a ref', () => {
        const result = renderHook(() => useSketchRef());

        expect(result.result.current).toEqual({ current: null });
    });
});
