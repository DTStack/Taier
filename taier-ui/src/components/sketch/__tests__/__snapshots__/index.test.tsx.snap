// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Sketch Component Should match snapshot 1`] = `
<body>
  <div>
    <div
      class="dt-sketch"
    >
      <div
        class="dt-sketch-header"
      >
        <form
          autocomplete="off"
          class="ant-form ant-form-inline"
        >
          <div
            class="ant-form-item"
          >
            <div
              class="ant-row ant-form-item-row"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <span
                      class="ant-input-affix-wrapper"
                      style="width: 210px;"
                    >
                      <input
                        class="ant-input"
                        id="name"
                        placeholder="按任务名称搜索"
                        type="text"
                        value=""
                      />
                      <span
                        class="ant-input-suffix"
                      >
                        <span
                          aria-label="search"
                          class="anticon anticon-search"
                          role="img"
                          style="cursor: pointer;"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="search"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                            />
                          </svg>
                        </span>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item"
          >
            <div
              class="ant-row ant-form-item-row"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <span
                      class="ant-input-affix-wrapper dt-multiple"
                    >
                      <input
                        class="ant-input"
                        id="multipleName"
                        placeholder="按任务名称搜索"
                        type="text"
                        value=""
                      />
                      <span
                        class="ant-input-suffix"
                      >
                        <div
                          style="display: flex;"
                        >
                          <div
                            class="suffix-item"
                            style="background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAMAAABEpIrGAAAAwFBMVEUAAAD///+AgICioqKSkpKcnJyUlJSbm5uVlZWXl5eWlpaWlpaYmJiZmZmWlpaVlZWVlZWWlpaVlZWYmJiWlpaYmJiWlpaXl5eYmJiXl5eXl5eWlpaYmJiYmJiXl5eYmJiYmJiWlpaXl5eXl5eWlpaWlpaXl5eWlpaWlpaYmJiXl5eXl5eYmJiYmJiXl5eYmJiXl5eWlpaXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eIXQHmAAAAP3RSTlMAAQILDhITFxggIi4vMjM1Oj9BQ0RISVFUXWBmam9zdHmBhIWGjZaXoaOkp6qvvL7EyMzR2t3g4eTn6erz+PmloSGXAAAAAWJLR0QB/wIt3gAAAJlJREFUOMtjYBgF1AUqdvYowE4ZTYEdK4QW1oPQbLZoCuyhNJ82mgCKAh1TRgwdyHw2M31BIMUtxo5DgayqpAYDg5yBsjknFgX2JgzGPMxWLAxcAuL6kpb2WEzgtzEyspZmUNOVMZDCaoW6PAODiCGDqSi/BXYFChwMDExKzEJaihJ8OLwJBLyaOLxpx0YgJJXR40JpNH2SBgAKqha54oD4rgAAAABJRU5ErkJggg==);"
                          />
                          <div
                            class="suffix-item"
                            style="background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAABmJLR0QA/wD/AP+gvaeTAAAB20lEQVRYhe2QPYgTURSFz33zJtkogrI6hcUqIm5pobiVVlsq6SwEmyQwZMCfThYRXiXYSZjMzdOwYVER0gsWay2ynYiFICgoGIw2oilm8q7NBpYVMkEN28xX3nM553CAgoKCgoIpMPPLOI4X55mhcvSjvu/v38sCc2emAsYY1ev1Fnbfmfk1M2/NtYCIlIMgeDYaja7tCj8HoCQii51O5/ROrd/ve0mSnLDWHv4fBZ4qpcae5/WSJDlkjNHbUo2INgA8EpHa5D+O42PD4fAtgA3n3CtmNv9UAMAZEbngnHtHRO+DILhnrd0H4DIRPVFKPQZwpdVqlQFAa30EwK0ois5rrVcB3JxmrqeJACAiz5VSZSK6GIbhLwBg5qsAlIjc3n5b8H2/CqCfZdlXz/PuMvNalmUZgAPT/HMXIKImgA8isrbjXCeiB865TefcJoB1IqoDgFLqDoCtZrO5QkQ38vxzF1BKuTAMa8YYBQDtdvukiJwtlUqX6vX6DwDodrsv0jT9ZK1dGo/Hb0TkOjMfFJHjANxU/7wCE4wxE6MvRLQ8CQeARqPx3Tm3XKlUvkVRdN/zvKpz7uFgMKhqrU/NmvEHzPzRWrv01wYzkLfA5zRNf86zQEFBQcGe8xvNVbTHyF7uqwAAAABJRU5ErkJggg==);"
                          />
                          <div
                            class="suffix-item"
                            style="background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAABmJLR0QA/wD/AP+gvaeTAAAB9klEQVRYhe2UP4gTQRjF37fJJqcRDw0kEUTwDygIIgh2NoIoWFwjKU5QDIRjJhxoddptY32wxcyuWEi4QnK1eJ1a2HqNIAh3V6jkApcyISCzz0YhitmcSqz2B9PM+2bem2+GATIyMjIyZkwURctRFC1P0vOzDkCynKZ7sw4wjakBrLUPrLU0xlz4k43b7XZpP3X7uYIGyQ0RaQC4/0u4inPuQKVS+Vyv192P+TiOjw0Gg01jzM1pm6d2wFp7GcCc7/sawGIYhsUx7QmAt57nrfX7/c0wDA8DgDHmiHNuRHJFRDYAXPzrAAAaANrNZnMHwAff9xcAoNPpFEhulcvl81rrKyKy5/v+NWPMoohsfR+rAI4CWEgzkElCHMcHkyTpAngtIrskLwHYU0rdICnW2scicp3kVxE5CWBFKfVsrENLAB6RfCMiO0qp4Hc+E98AyVsAPpJcIwkReUXyaRzHJ4wxxz3Pu+153plut+uq1eq78bXGmFMAHpK8KiJ30jow8QpINkjGWut1rfW6Uuq5iLxIkuQugE8ADjnnbK1WeykiPx1Ea73d6/VOa62308xTO5DP5+/Nz89/GZ8bjUZLhUJhrtVq7YZheK5YLJ4dDofvc7mcXyqVRuO1QRAk08z/C9bawFobTNJn/hWLSH/WHhkZGRn/xDfA/cJ9mTtA3QAAAABJRU5ErkJggg==);"
                          />
                        </div>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item"
          >
            <div
              class="ant-row ant-form-item-row"
            >
              <div
                class="ant-col ant-form-item-label"
              >
                <label
                  class=""
                  for="owner"
                  title="操作人"
                >
                  操作人
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <div
                      class="ant-select ant-select-in-form-item dt-form-shadow-bg ant-select-single ant-select-allow-clear ant-select-show-arrow ant-select-show-search"
                      style="width: 200px;"
                    >
                      <div
                        class="ant-select-selector"
                      >
                        <span
                          class="ant-select-selection-search"
                        >
                          <input
                            aria-activedescendant="owner_list_0"
                            aria-autocomplete="list"
                            aria-controls="owner_list"
                            aria-haspopup="listbox"
                            aria-owns="owner_list"
                            autocomplete="off"
                            class="ant-select-selection-search-input"
                            id="owner"
                            role="combobox"
                            type="search"
                            value=""
                          />
                        </span>
                        <span
                          class="ant-select-selection-placeholder"
                        >
                          请选择操作人
                        </span>
                      </div>
                      <span
                        aria-hidden="true"
                        class="ant-select-arrow"
                        style="user-select: none;"
                        unselectable="on"
                      >
                        <span
                          aria-label="down"
                          class="anticon anticon-down ant-select-suffix"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="down"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item"
          >
            <div
              class="ant-row ant-form-item-row"
            >
              <div
                class="ant-col ant-form-item-label"
              >
                <label
                  class=""
                  for="rangeDate"
                  title="日期范围选择"
                >
                  日期范围选择
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <div
                      class="ant-picker ant-picker-range"
                    >
                      <div
                        class="ant-picker-input ant-picker-input-active"
                      >
                        <input
                          autocomplete="off"
                          id="rangeDate"
                          placeholder="Start date"
                          readonly=""
                          size="12"
                          value=""
                        />
                      </div>
                      <div
                        class="ant-picker-range-separator"
                      >
                        <span
                          aria-label="to"
                          class="ant-picker-separator"
                        >
                          <span
                            aria-label="swap-right"
                            class="anticon anticon-swap-right"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="swap-right"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="0 0 1024 1024"
                              width="1em"
                            >
                              <path
                                d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                      <div
                        class="ant-picker-input"
                      >
                        <input
                          autocomplete="off"
                          placeholder="End date"
                          readonly=""
                          size="12"
                          value=""
                        />
                      </div>
                      <div
                        class="ant-picker-active-bar"
                        style="left: 0px; width: 0px; position: absolute;"
                      />
                      <span
                        class="ant-picker-suffix"
                      >
                        <span
                          aria-label="calendar"
                          class="anticon anticon-calendar"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="calendar"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item"
          >
            <div
              class="ant-row ant-form-item-row"
            >
              <div
                class="ant-col ant-form-item-label"
              >
                <label
                  class=""
                  for="date"
                  title="业务日期"
                >
                  业务日期
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <div
                      class="ant-picker"
                      style="width: 210px;"
                    >
                      <div
                        class="ant-picker-input"
                      >
                        <input
                          autocomplete="off"
                          id="date"
                          placeholder="业务日期"
                          readonly=""
                          size="12"
                          title=""
                          value=""
                        />
                        <span
                          class="ant-picker-suffix"
                        >
                          <span
                            aria-label="calendar"
                            class="anticon anticon-calendar"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="calendar"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item"
          >
            <div
              class="ant-row ant-form-item-row"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <div
                      class="ant-select ant-select-in-form-item ant-select-single ant-select-show-arrow"
                      style="width: 150px;"
                    >
                      <div
                        class="ant-select-selector"
                      >
                        <span
                          class="ant-select-selection-search"
                        >
                          <input
                            aria-activedescendant="select_list_0"
                            aria-autocomplete="list"
                            aria-controls="select_list"
                            aria-haspopup="listbox"
                            aria-owns="select_list"
                            autocomplete="off"
                            class="ant-select-selection-search-input"
                            id="select"
                            readonly=""
                            role="combobox"
                            style="opacity: 0;"
                            type="search"
                            unselectable="on"
                            value=""
                          />
                        </span>
                        <span
                          class="ant-select-selection-placeholder"
                        />
                      </div>
                      <span
                        aria-hidden="true"
                        class="ant-select-arrow"
                        style="user-select: none;"
                        unselectable="on"
                      >
                        <span
                          aria-label="down"
                          class="anticon anticon-down ant-select-suffix"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="down"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item"
          >
            <div
              class="ant-row ant-form-item-row"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <div
                      class="ant-radio-group ant-radio-group-outline"
                      id="radioGroup"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
        <div
          class="dt-sketch-extra"
        >
          <button>
            test
          </button>
        </div>
      </div>
      <div
        class="dt-sketch-header-bar"
      />
      <div
        class="ant-table-wrapper dt-sketch-table"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup>
                      <col
                        class="ant-table-selection-col"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell ant-table-selection-column"
                        >
                          <div
                            class="ant-table-selection"
                          >
                            <label
                              class="ant-checkbox-wrapper"
                            >
                              <span
                                class="ant-checkbox"
                              >
                                <input
                                  aria-label="Select all"
                                  class="ant-checkbox-input"
                                  type="checkbox"
                                  value=""
                                />
                                <span
                                  class="ant-checkbox-inner"
                                />
                              </span>
                            </label>
                          </div>
                        </th>
                        <th
                          class="ant-table-cell"
                        />
                        <th
                          class="ant-table-cell"
                        />
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                      >
                        <td
                          class="ant-table-cell ant-table-selection-column"
                        >
                          <label
                            class="ant-checkbox-wrapper"
                          >
                            <span
                              class="ant-checkbox"
                            >
                              <input
                                class="ant-checkbox-input"
                                type="checkbox"
                                value=""
                              />
                              <span
                                class="ant-checkbox-inner"
                              />
                            </span>
                          </label>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          1
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          test1
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="2"
                      >
                        <td
                          class="ant-table-cell ant-table-selection-column"
                        >
                          <label
                            class="ant-checkbox-wrapper"
                          >
                            <span
                              class="ant-checkbox"
                            >
                              <input
                                class="ant-checkbox-input"
                                type="checkbox"
                                value=""
                              />
                              <span
                                class="ant-checkbox-inner"
                              />
                            </span>
                          </label>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          2
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          test2
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
              <div
                class="ant-table-footer"
              >
                <div
                  class="flex-between"
                >
                  <div
                    style="padding-left: 25px;"
                  />
                  <div>
                    <ul
                      class="ant-pagination"
                    >
                      <li
                        aria-disabled="true"
                        class="ant-pagination-prev ant-pagination-disabled"
                        title="Previous Page"
                      >
                        <button
                          class="ant-pagination-item-link"
                          disabled=""
                          tabindex="-1"
                          type="button"
                        >
                          <span
                            aria-label="left"
                            class="anticon anticon-left"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="left"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                              />
                            </svg>
                          </span>
                        </button>
                      </li>
                      <li
                        class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                        tabindex="0"
                        title="1"
                      >
                        <a
                          rel="nofollow"
                        >
                          1
                        </a>
                      </li>
                      <li
                        aria-disabled="true"
                        class="ant-pagination-next ant-pagination-disabled"
                        title="Next Page"
                      >
                        <button
                          class="ant-pagination-item-link"
                          disabled=""
                          tabindex="-1"
                          type="button"
                        >
                          <span
                            aria-label="right"
                            class="anticon anticon-right"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="right"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                              />
                            </svg>
                          </span>
                        </button>
                      </li>
                      <li
                        class="ant-pagination-options"
                      >
                        <div
                          aria-label="Page Size"
                          class="ant-select ant-pagination-options-size-changer ant-select-single ant-select-show-arrow"
                        >
                          <div
                            class="ant-select-selector"
                          >
                            <span
                              class="ant-select-selection-search"
                            >
                              <input
                                aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                                aria-autocomplete="list"
                                aria-controls="rc_select_TEST_OR_SSR_list"
                                aria-expanded="false"
                                aria-haspopup="listbox"
                                aria-label="Page Size"
                                aria-owns="rc_select_TEST_OR_SSR_list"
                                autocomplete="off"
                                class="ant-select-selection-search-input"
                                id="rc_select_TEST_OR_SSR"
                                readonly=""
                                role="combobox"
                                style="opacity: 0;"
                                type="search"
                                unselectable="on"
                                value=""
                              />
                            </span>
                            <span
                              class="ant-select-selection-item"
                              title="20 / page"
                            >
                              20 / page
                            </span>
                          </div>
                          <span
                            aria-hidden="true"
                            class="ant-select-arrow"
                            style="user-select: none;"
                            unselectable="on"
                          >
                            <span
                              aria-label="down"
                              class="anticon anticon-down ant-select-suffix"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
